src/main/frontend/components/ReviewTile.js

```diff
@@ -12,13 +19,80 @@ const ReviewTile = (props) => {
     userReview = "User Review: "
   }
 
+  const voted = count => {
+    let localCount = stateCount
+    if (downvoted == true && count == 1 && upvoted == false) {
+      setDownvoted(false)
+      setUpvoted(true)
+      localCount += 2
+    } else if (upvoted == true && count == -1 && downvoted == false) {
+      setUpvoted(false)
+      setDownvoted(true)
+      localCount -= 2
+    } else if (count == 1 && upvoted == true) {
+      setUpvoted(false)
+      setDownvoted(false)
+      localCount -= 1
+    } else if (count == 1 && upvoted == false) {
+      setUpvoted(true)
+      setDownvoted(false)
+      localCount += 1
+    } else if (count == -1 && downvoted == true) {
+      setDownvoted(false)
+      setUpvoted(false)
+      localCount += 1
+    } else if (count == -1 && downvoted == false) {
+      setDownvoted(true)
+      setUpvoted(false)
+      localCount -= 1
+    }
+    setCount(localCount)
+    updateVoteCount(id, localCount)

```

Comment:
Instead of setting the state on 49 before the fetch request has been made, we should let the response of the fetch request inform how to update state.  Ideally it will send back the new review vote score that we can set in state